### PR TITLE
fix rsync output utf8decode bug

### PR DESCRIPTION
**Pull request by intspt**
fix issue: https://github.com/flutter/flutter/issues/81666
The rsync command will be garbled when outputting Chinese, so an error will be reported when it is finally parsed with UTF8.
BTW, It is recommended to add the parameter -8 to all the places where rsync is used to avoid possible UTF8 decoding failure.
image
image

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter_tools/lib/src/ios/mac.dart
Comment (jmagman) [inline packages/flutter_tools/lib/src/ios/mac.dart:465]: Can you add a comment about why we added this?  Staring at the `rsync` man page I don't think it's super obvious why we would use this flag.  Something like:
```suggestion
            '-8', // Avoid mangling filenames with encodings that do not match the current locale.
```

**Review by jmagman — CHANGES_REQUESTED** [review]
Thanks for tracking this down!  Minor nit for adding a comment.

I audited the other places we `rsync` and I think the following could also use this flag:
This one is our generic "unzip" code, which is used for unzipping a pre-built iOS/macOS bundles, which could contain Chinese or other valid encoded file names:
https://github.com/flutter/flutter/blob/5b6de035ce3c43d3865243f6cebe6dcb977d8e66/packages/flutter_tools/lib/src/base/os.dart#L455
https://github.com/flutter/flutter/blob/5b6de035ce3c43d3865243f6cebe6dcb977d8e66/packages/flutter_tools/lib/src/ios/application_package.dart#L35-L37
https://github.com/flutter/flutter/blob/5b6de035ce3c43d3865243f6cebe6dcb977d8e66/packages/flutter_tools/lib/src/macos/application_package.dart#L67

______
And this one could be problematic for embedded dart-side assets file names:
https://github.com/flutter/flutter/blob/2d07436dbd535fa545eb3bae8c399a071c36ec93/packages/flutter_tools/bin/xcode_backend.dart#L191-L198

If you're willing to update those spots (and tests), that would be super helpful!  If not, you can update this PR with a comment and I'll merge it.

**Comment by intspt** [comment]
Hi, I updated it as you suggested.

**Review by jmagman — APPROVED** [review]
I wasn't able to reproduce the original issue, but I did try it to validate it didn't cause (obvious) regressions to resources without UTF-8 issues.

**Comment by jmagman** [comment]
+ 2nd reviewer @christopherfujino

**Comment by intspt** [comment]
> I wasn't able to reproduce the original issue, but I did try it to validate it didn't cause (obvious) regressions to resources without UTF-8 issues.

Just add a picture with Chinese name to the Runner project.

**Comment by jmagman** [comment]
> Just add a picture with Chinese name to the Runner project.

I did, it runs fine.  This may be related to locale settings?
Screen Shot 2022-01-12 at 11 14 52 AM
Screen Shot 2022-01-12 at 11 15 38 AM

I'm not worried about reproducing it to validate this change, since you can reproduce.  We'll get further verification from other reporters once we get a second review from @christopherfujino and this merges.

**Comment by intspt** [comment]
> > Just add a picture with Chinese name to the Runner project.
> 
> I did, it runs fine. This may be related to locale settings? Screen Shot 2022-01-12 at 11 14 52 AM Screen Shot 2022-01-12 at 11 15 38 AM
> 
> I'm not worried about reproducing it to validate this change, since you can reproduce. We'll get further verification from other reporters once we get a second review from @christopherfujino and this merges.

use flutter run. And it's only the first time that this problem occurs

**Review by christopherfujino — APPROVED** [review]
LGTM, thanks for this fix!

**Comment by ahyangnb** [comment]
thanks for this fix!

**Pull request merged**
